trading/trader/order.py

```diff
@@ -18,54 +18,60 @@
 import math
 from abc import *
 from asyncio import Lock
+from dataclasses import dataclass, field
+from typing import List, Any
+
 from tools.logging.logging_util import get_logger
 
 from tools.symbol_util import split_symbol
 from config import TradeOrderSide, OrderStatus, TraderOrderType, SIMULATOR_LAST_PRICES_TO_CHECK, \
     ExchangeConstantsTickersColumns as eC, FeePropertyColumns, ExchangeConstantsMarketPropertyColumns
+from trading.exchanges.exchange_dispatcher import ExchangeDispatcher
 
 """ Order class will represent an open order in the specified exchange
 In simulation it will also define rules to be filled / canceled
 It is also use to store creation & fill values of the order """
 
 
+@dataclass
 class Order:
     __metaclass__ = ABCMeta
 
-    def __init__(self, trader):
-        super().__init__()
-        self.trader = trader
+    trader: Any = field(init=True, repr=False)
+    exchange: ExchangeDispatcher = field(init=False, repr=False)
+    is_simulated: bool = field(init=False, repr=False)
+    side: TradeOrderSide = None
+    symbol: str = None
+    origin_price: float = 0
+    origin_stop_price: float = 0
+    origin_quantity: float = 0
+    market_total_fees: float = 0
+    filled_quantity: float = 0
+    filled_price: float = 0
+    fee: float = 0
+    currency: str = None
+    market: str = None
+    order_id: str = None
+    status: OrderStatus = OrderStatus.OPEN
+    order_type: Any = None
+    timestamp: int = None
+    creation_time: int = time.time()
+    canceled_time: int = 0
+    executed_time: int = 0
+    last_prices: List[Any] = None
+    created_last_price: float = 0
+    order_profitability: float = 0
+    linked_to: Any = None
+    is_from_this_octobot: bool = True
+    linked_portfolio: Any = None
+    order_notifier: Any = None
+    linked_orders: List[Any] = List
+    taker_or_maker: ExchangeConstantsMarketPropertyColumns = None
+    lock: Lock = Lock()
```

Comment:
Looks like we can't initialize an asyncio.Lock in a dataclass that way (from CI)